### PR TITLE
Interpretations: add `aggregation_strategy`, better docstring and corner cases handling

### PR DESCRIPTION
**Pull request by AntoninPoche**
## Description

This PR introduces the `aggregation_strategy` parameter to interpretation methods. It is helpful to interpret the concepts' spaces constructed with an aggregation. In particular, when the granularity is `SAMPLE`.

I needed this to interpret probes on sample-level annotations.

Furthermore, this PR improves the docstrings, addresses some corner cases, and resolves a bug in interpretation from the vocabulary.

No test was added, because this is inherently tested when the granularity is larger than `TOKEN` and both `SENTENCE` and `SAMPLE` are tested. Hence, we know it works for the `MEAN` aggregation, and as others have been unitarily extensively tested, I think that no additional tests are required.

## Type of Change

- 📚 Examples / docs / tutorials / dependencies update
- 🔧 Bug fix (non-breaking change which fixes an issue)
- 🥂 Improvement (non-breaking change which improves an existing feature)

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/FOR-sight-ai/interpreto/blob/main/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/FOR-sight-ai/interpreto/blob/main/docs/contributing.md) guide.
- [x] I've successfully run the style checks using `make lint`.
- [x] I've written tests for all new methods and classes that I created and successfully ran `make test`.
- [x] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: interpreto/concepts/interpretations/base.py [hunk 1]
```diff
@@ -395,25 +405,56 @@ def concepts_activations_from_vocab(
...
+            # if we are not in a template [CLS] a [EOS]
+            if len(template_ids) != 3:  # type: ignore
+                warnings.warn(
+                    "When tokenizing a single character, the provided model does not output 3 token ids. "
+                    "Our implementation assumes that the model outputs is [CLS] a [EOS]. "
+                    "Indeed, when `aggregation_strategy` is `CLS_TOKEN`, the first token is considered as the CLS token. "
+                    "If the [CLS] token is still the first token, you can ignore this warning. "
+                    "Otherwise, either choose another model or contact the developers to find a workaround.",
+                    stacklevel=2,
+                )
```
Comment (fanny-jourdan) [inline interpreto/concepts/interpretations/base.py:440]: Not sure to understand this

Comment (AntoninPoche) [reply interpreto/concepts/interpretations/base.py:440]: In some cases, I assume a format; when the format does correspond, I raise a warning.

It implies that when we tokenize "a", we expect three tokens: ["[CLS]", "a", "[EOS]"], which are actually their token IDs. In anycase, the important thing is that the first token is the [CLS] token.
